lib/std/tz.zig

```diff
@@ -0,0 +1,233 @@
+const std = @import("std.zig");
+const builtin = @import("builtin");
+
+pub const Transition = struct {
+    ts: i64,
+    timetype: *Timetype,
+};
+
+pub const Timetype = struct {
+    offset: i32,
+    flags: u8,
+    name_data: [6:0]u8,
+
+    pub fn name(self: Timetype) [:0]const u8 {
+        return std.mem.sliceTo(self.name_data[0..], 0);
+    }
+
+    pub fn isDst(self: Timetype) bool {
+        return (self.flags & 0x01) > 0;
+    }
+
+    pub fn standardTimeIndicator(self: Timetype) bool {
+        return (self.flags & 0x02) > 0;
+    }
+
+    pub fn utIndicator(self: Timetype) bool {
+        return (self.flags & 0x04) > 0;
+    }
+};
+
+pub const Leapsecond = struct {
+    occurrence: i48,
+    correction: i16,
+};
+
+pub const Tz = struct {
+    allocator: std.mem.Allocator,
+    transitions: []const Transition,
+    timetypes: []const Timetype,
+    leapseconds: []const Leapsecond,
+    footer: []const u8,
+
+    pub fn parse(allocator: std.mem.Allocator, reader: anytype) !Tz {
+        const Header = extern struct {
+            magic: [4]u8,
+            version: u8,
+            reserved: [15]u8,
+        };
+
+        const Counts = extern struct {
+            isutcnt: u32,
+            isstdcnt: u32,
+            leapcnt: u32,
+            timecnt: u32,
+            typecnt: u32,
+            charcnt: u32,
+        };
+
+        // Parse and skip the legacy header and data
+        {
+            const header = try reader.readStruct(Header);
+            if (!std.mem.eql(u8, &header.magic, "TZif")) return error.BadHeader;
+            if (header.version == 0) return error.UnsupportedLegacyFormat;
```

Comment:
Why not support version 0? it's quite trivial and IIRC I ran into it on a few systems.

Reply:
I imagined that it would cause an entire new class of possible errors for the future date library, e.g. "You've tried to process a time after 2038" failures that need to be taken into account.

If there actually are systems that only use the legacy format then I guess we should support it as well though; at least until 2038...

Reply:
Added legacy file support.